File: app/pages/repos/_repo_id/models/_model_id/production_predictions/index/props.ts
import { ModelLayoutInfo } from 'layouts/model_layout'

export type Props = {
	modelLayoutInfo: ModelLayoutInfo
	pagination: {
		after: string | null
		before: string | null
	}
	predictionTable: { rows: PredictionTableRow[] }
}

type PredictionTableRow = {
	date: string
	identifier: string
	output: string
}

File: app/pages/repos/_repo_id/models/_model_id/production_predictions/index/props.rs
use tangram_app_layouts::model_layout::ModelLayoutInfo;

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Props {
	pub model_layout_info: ModelLayoutInfo,
	pub prediction_table: Option<PredictionTable>,
	pub pagination: Pagination,
}

#[derive(serde::Serialize, Debug)]
pub struct PredictionTable {
	pub rows: Vec<PredictionTableRow>,
}

#[derive(serde::Serialize, Debug)]
pub struct PredictionTableRow {
	pub date: String,
	pub identifier: String,
	pub output: String,
}

#[derive(serde::Serialize, Debug)]
pub struct Pagination {
	pub after: Option<usize>,
	pub before: Option<usize>,
}

#[derive(serde::Serialize, Debug)]
pub struct PaginationRange {
	pub start: usize,
	pub end: usize,
	pub total: usize,
}
